rotkehlchen/db/updates.py

```diff
@@ -80,6 +82,45 @@ def _get_remote_info_json(self) -> dict[str, Any]:
 
         return json_data
 
+    def update_rpc_nodes(self, local_version: int, to_version: int) -> None:
+        """
+        Check the updates in the inclusive range [local_version + 1, to_version] and update the
+        rpc nodes for those versions.
+        """
+        for version in range(local_version + 1, to_version + 1):
+            file_url = RPC_NODES_URL.format(branch=self.branch, version=version)
+            try:
+                updates = query_file(file_url, is_json=True)
+            except RemoteError as e:
+                log.warning(f'Failed to update rpc nodes due to {str(e)}')
+                break
+
+            updated_rpc_nodes = updates.get('rpc_nodes')
+            if updated_rpc_nodes is None:
+                log.error(f'Remote update {file_url} does not contain rpc_nodes key')
+                break
+
+            log.info(f'Applying update for rpc nodes from v{version}')
+            with (
+                GlobalDBHandler().conn.read_ctx() as cursor,
+                GlobalDBHandler().conn.write_ctx() as write_cursor,
+            ):
+                cursor.execute('DELETE FROM rpc_nodes')
+                write_cursor.executemany(
+                    'INSERT OR IGNORE INTO rpc_nodes(name, endpoint, owned, active, weight, blockchain) '  # noqa: E501
+                    'VALUES (?, ?, ?, ?, ?, ?)',
+                    [
+                        (node['name'], node['endpoint'], node['owned'], node['active'], str(FVal(node['weight'])), node['blockchain'])  # noqa: E501
+                        for node in updated_rpc_nodes
+                    ],
```

Comment:
This can raise InputError I believe (please check it) if we make a mistake with one of the updates and the values don't have the correct type, please handle it and any possible keyError.